File: example/spike/hello/hello.c
/*
 * Copyright 2021, Breakaway Consulting Pty. Ltd.
 * Copyright 2022, UNSW (ABN 57 195 873 179)
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */
#include <stdint.h>
#include <microkit.h>

void
init(void)
{
    microkit_dbg_puts("hello, world\n");
}

void
notified(microkit_channel ch)
{
}

File: example/rust/hello.rs
//
// Copyright 2025, UNSW
//
// SPDX-License-Identifier: BSD-2-Clause
//

#![no_std]
#![no_main]

use sel4_microkit::{NullHandler, protection_domain, debug_println};

#[protection_domain()]
fn init() -> NullHandler {
    debug_println!("hello, world from Rust!");

    NullHandler::new()
}
